cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go

```diff
@@ -31,9 +31,11 @@ const (
 	// MaxFreeDifferenceRatio describes how free resources (allocatable - daemon and system pods)
 	// can differ between groups in the same NodeGroupSet
 	MaxFreeDifferenceRatio = 0.05
-	// MaxMemoryDifferenceInKiloBytes describes how much memory
-	// capacity can differ but still be considered equal.
-	MaxMemoryDifferenceInKiloBytes = 256000
+)
+
+var (
+	// MaxMemoryDifference describes how much memory capacity can differ but still be considered equal.
+	MaxMemoryDifference = resource.MustParse("256Mi")
```

Comment:
How big is the diff we are seeing in real nodes?
Tolerating 256Mi seems too much as to consider nodeGroups equal. The original intention was to tolerate 128Ki.
I think this should be 128Ki https://github.com/kubernetes/autoscaler/commit/e8b3c2a111eb9b3b4fe15b4d4081175267ff6d76

Reply:
Please see the test case I've added below which came from a real world test case. The values that came through the code (via much debug logging) were `16116152Ki` and `15944120Ki`, which is `168Mi`, just over a 1% difference in this case

Please also review the attached BZ which has more details from a customer who report differences in a similar magnitude.

Reply:
Wouldn't this `MaxMemoryDifference` also apply to much more smaller instances to the point of making the check loosing its value?
i.e If the possible diff range increase with the instance size, should we may be make our tolerance window a percentage of the given total size?

Reply:
Let's keep the discussion to the upstream PR for better visibility https://github.com/kubernetes/autoscaler/pull/3124/files#r422931565